sky/utils/kubernetes_utils.py

```diff
@@ -147,6 +159,89 @@ def get_kubernetes_nodes() -> List[Any]:
     return nodes
 
 
+def check_instance_fits(instance: str) -> Tuple[bool, Optional[str]]:
+    """Checks if the instance fits on the Kubernetes cluster.
+
+    If the instance has GPU requirements, checks if the GPU type is
+    available on the cluster and if enough CPU/memory is available on any node
+    with the GPU type.
+
+    Args:
+        instance: str, the instance type to check.
+
+    Returns:
+        bool: True if the instance fits on the cluster, False otherwise.
+        Optional[str]: Error message if the instance does not fit.
+    """
+
+    def check_cpu_mem_fits(
+            candidate_instance_type: kubernetes_cloud.KubernetesInstanceType,
```

Comment:
nit: Can we move the `KubernetesInstanceType` to this utils file, so we can import this module in the `clouds.Kubernetes` instead of the other way around?

Not feeling strongly about this, just that it feels a bit surprising to me to see the cloud being imported in the utils module.

Reply:
Sounds good!